pkg/operator/ceph/disruption/nodedrain/reconcile.go

```diff
@@ -226,13 +235,14 @@ func (r *ReconcileNode) reconcile(request reconcile.Request) (reconcile.Result,
 		// delete the canary deployments that aren't triggered by drains, but are on nodes that aren't occupied by OSDs
 		// if it's on a draining node, we don't want to kill the canary.
 		if canarayDeployment.Status.ReadyReplicas > 0 {
+			logger.Infof("deleting drain-canary deployment on node %s as OSDs are no longer scheduled here", request.Name)
 			err := r.client.Delete(context.TODO(), canarayDeployment)
 			if err != nil {
 				return reconcile.Result{}, errors.Wrapf(err, "could not delete deployment %q", key)
 			}
 		}
 	}
-	return reconcile.Result{}, nil
+	return reconcile.Result{Requeue: true, RequeueAfter: time.Minute * 2}, nil
```

Comment:
Why do we always requeue after 2 min?

Reply:
It's not strictly necessary, but it will ensure accurate state if any changes happen that don't trigger reconciliation.

It's not an expensive op, but we could get rid of it as I don't know if there are any more such scenarios. The other changes in this patch address one such scenario I missed  i.e. when the node object is deleted, but the canary doesn't get deleted.

Reply:
How did we re-trigger updates previously? Did we only rely on certain events such as CR changes? If this is just to periodically check in case there are any updates, I'd suggest a longer time such as 15 minutes.

Reply:
Yeah, it was based on multiple types of CR changes.
I'll update to 15 minutes.